apps/client/src/components/Questions.js

```diff
@@ -66,120 +153,99 @@ const Questions = ({
-          <Question
-            question={q}
-            onGoToPrev={onQuestionPrev}
-            onGoToNext={onQuestionNext}
-            saveAnswer={saveAnswer}
-            showNext
-            showPrev
-            {...{
-              checker,
-              questionIndex,
-              questionNeedsPermit,
-              userAnswer,
-            }}
-          />
-        ) : (
-          // Show the answer with an edit button
-          <QuestionAnswer
-            onClick={() => onGoToQuestion(i)}
-            {...{ questionNeedsPermit, userAnswer }}
-          />
-        )}
-      </StepByStepItem>
-    );
-  });
+  return (
+    <>
+      {checker.stack.map((q, i) => {
+        // Define userAnswer
+        const booleanAnswers =
+          !q.options && booleanOptions.find((o) => o.value === q.answer);
+        const userAnswer = q.options ? q.answer : booleanAnswers?.label;
+
+        // Define if question is the current one
+        const isCurrentQuestion =
+          q === checker.stack[questionIndex] && isActive("questions");
+
+        // Hide unanswered questions (eg: on browser refresh)
+        if (!isCurrentQuestion && !userAnswer) return null;
+
+        // Check if currect question is causing a permit requirement
+        const questionNeedsPermit = !!permitsPerQuestion[i];
+
+        return (
+          <StepByStepItem
+            active={isCurrentQuestion}
+            checked={userAnswer}
+            customSize
+            heading={q.text}
+            highlightActive={isCurrentQuestion}
+            key={`question-${q.id}-${i}`}
+            style={isCurrentQuestion ? activeStyle : {}}
+          >
+            {isCurrentQuestion ? (
+              // Show the current question
+              <Question
+                question={q}
+                onGoToPrev={onQuestionPrev}
+                onGoToNext={onQuestionNext}
+                saveAnswer={saveAnswer}
+                showNext
+                showPrev
+                {...{
+                  checker,
+                  questionIndex,
+                  questionNeedsPermit,
+                  userAnswer,
+                }}
+              />
+            ) : (
+              // Show the answer with an edit button
+              <QuestionAnswer
+                onClick={() => onGoToQuestion(i)}
+                {...{ questionNeedsPermit, userAnswer }}
+              />
+            )}
+          </StepByStepItem>
+        );
+      })}
+      {checker._getUpcomingQuestions().map((q, i) => {
+        // @TODO: refactor this part
+        // Define userAnswer
+        const booleanAnswers =
+          !q.options && booleanOptions.find((o) => o.value === q.answer);
+        const userAnswer = q.options ? q.answer : booleanAnswers?.label;
+
+        // Skip unanswered questions
+        if (!userAnswer) return null;
+
+        // Check if currect question is causing a permit requirement
+        const questionNeedsPermit = !!permitsPerQuestion[i];
```

Comment:
Why do we need `i` here, and index below and `index` by `goToQuestion. `

Reply:
Good point! Fixed this